### PR TITLE
Enable `gardener-resource-manager`'s `TokenInvalidator` and `ProjectedTokenMount` webhooks in seed and shoot

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security
/kind enhancement
/merge squash

**What this PR does / why we need it**:
This PR enables the `gardener-resource-manager`'s `TokenInvalidator` (#4817) and the `ProjectedTokenMount` (#4873) webhooks for the seed and shoot clusters.

This is a prerequisite to switch all our components from static `ServiceAccount` tokens to projected token mounts (auto-rotated and expiring).

**Which issue(s) this PR fixes**:
Part of #4878

**Special notes for your reviewer**:
In order to test the change you need to use the `rfranzke/gardener-resource-manager:v1.36.0-dev-grm-webhooks` image for `gardener-resource-manager` (you have to overwrite it in your local `charts/images.yaml`).

**Release note**:

```noteworthy operator
The `gardener-resource-manager`'s `TokenInvalidator` and the `ProjectedTokenMount` webhooks are now enabled for the seed and shoot clusters.
```

### CODE REVIEW

File: pkg/operation/seed/components.go [hunk 1]
```diff
@@ -105,15 +105,17 @@ func defaultGardenerResourceManager(c client.Client, imageVector imagevector.Ima
...
 
 	gardenerResourceManager := resourcemanager.New(c, v1beta1constants.GardenNamespace, image.String(), 3, resourcemanager.Values{
-		ConcurrentSyncs:                     pointer.Int32(20),
-		MaxConcurrentRootCAPublisherWorkers: pointer.Int32(20),
-		HealthSyncPeriod:                    utils.DurationPtr(time.Minute),
-		ResourceClass:                       pointer.String(v1beta1constants.SeedResourceManagerClass),
-		SyncPeriod:                          utils.DurationPtr(time.Hour),
+		ConcurrentSyncs:                      pointer.Int32(20),
+		MaxConcurrentTokenInvalidatorWorkers: pointer.Int32(20),
+		MaxConcurrentRootCAPublisherWorkers:  pointer.Int32(20),
```
Comment (timebertt) [inline pkg/operation/seed/components.go]: 5 should be enough for the new controllers.

File: pkg/operation/botanist/resource_manager.go [hunk 2]
```diff
@@ -45,16 +45,18 @@ func (b *Botanist) DefaultResourceManager() (resourcemanager.Interface, error) {
...
-		TargetDisableCache:                  pointer.Bool(true),
-		WatchedNamespace:                    pointer.String(b.Shoot.SeedNamespace),
+		AlwaysUpdate:                         pointer.Bool(true),
+		ClusterIdentity:                      b.Seed.GetInfo().Status.ClusterIdentity,
+		ConcurrentSyncs:                      pointer.Int32(20),
+		HealthSyncPeriod:                     utils.DurationPtr(time.Minute),
+		MaxConcurrentHealthWorkers:           pointer.Int32(10),
+		MaxConcurrentTokenInvalidatorWorkers: pointer.Int32(10),
+		MaxConcurrentTokenRequestorWorkers:   pointer.Int32(10),
+		MaxConcurrentRootCAPublisherWorkers:  pointer.Int32(10),
```
Comment (timebertt) [inline pkg/operation/botanist/resource_manager.go]: Same here, 5 should be enough

File: pkg/operation/botanist/component/resourcemanager/resource_manager.go [hunk 3]
```diff
@@ -736,8 +769,101 @@ func (r *resourceManager) emptyPodDisruptionBudget() *policyv1beta1.PodDisruptio
...
+
+	data, err := registry.AddAllAndSerialize(mutatingWebhookConfiguration)
+	if err != nil {
+		return err
+	}
+
+	return managedresources.CreateForShoot(ctx, r.client, r.namespace, ManagedResourceName, false, data)
+}
+
+func (r *resourceManager) getMutatingWebhookConfigurationWebhooks() []admissionregistrationv1.MutatingWebhook {
```
Comment (timebertt) [inline pkg/operation/botanist/component/resourcemanager/resource_manager.go]: It's probably a good idea to make the webhook config reusable between the component and envtest package. E.g. like it is done for gsac:
https://github.com/gardener/gardener/blob/aeffd6154d2cee2c84b36530d812d9d3a4ef59f4/pkg/operation/botanist/component/seedadmissioncontroller/seedadmissioncontroller.go#L330

**Review by timebertt — COMMENTED** [review]
Nits, rest looks good.

**Review by timebertt — APPROVED** [review]
Nice!
/lgtm

**Comment by rfranzke** [comment]
Thanks for the review @timebertt!
@BeckerMax Do you still plan to review this PR?

**Comment by BeckerMax** [comment]
/unassign 
no reason to block it because I am slowed due to other tasks. You can go ahead without my review.

**Comment by rfranzke** [comment]
Okay, thank you @BeckerMax, then let's do this :)

**Pull request merged**